### PR TITLE
Update release notes

### PR DESCRIPTION
**Pull request by vgvassilev**
@Teemperor fyi.

### CODE REVIEW

File: README/ReleaseNotes/v616/index.md [hunk 1]
```diff
@@ -57,6 +57,17 @@ They were deprecated before, or never ported from configure, make to CMake.
 
 It was never used in TVirtualX interfaces. If GLUtesselator forward declaration is required, use TGLUtil.h include instead. 
 
+## C++ Modules Techology Preview
```
Comment (Axel-Naumann) [inline README/ReleaseNotes/v616/index.md]: Tech*n*ology

File: README/CREDITS [hunk 2]
```diff
@@ -912,6 +917,11 @@ N: Matevz Tadel
 E: Matevz.Tadel@cern.ch
 D: Several contributions to the 3-d graphics and GL interface, WebEVE
 
+N: Yuka Takahashi
+E: Yuka.Takahashi@cern.ch
+D: runtime c++ modules integration in the CMS experiment
```
Comment (Axel-Naumann) [inline README/CREDITS]: This does not read like a ROOT contribution. Can you rephrase this to clarify the ROOT aspect of it? Maybe as simple as "integration of ROOT runtime modules in the CMS experiment"?

File: README/CREDITS [hunk 3]
```diff
@@ -864,6 +864,11 @@ N: Daniel Sigg
 E: sigg_d@ligo.caltech.edu
 D: TGNumberEntry GUI widget
 
+N: Oksana Shadura
+E: Oksana.Shadura@cern.ch
+D: runtime c++ modules
+D: contributions to cmake
```
Comment (Axel-Naumann) [inline README/CREDITS]: + rootbench?

Comment (vgvassilev) [reply README/CREDITS]: Yep.

File: README/CREDITS [hunk 4]
```diff
@@ -864,6 +864,12 @@ N: Daniel Sigg
 E: sigg_d@ligo.caltech.edu
 D: TGNumberEntry GUI widget
 
+N: Oksana Shadura
+E: Oksana.Shadura@cern.ch
```
Comment (yamaguchi1024) [inline README/CREDITS]: Not capital?

File: README/README.CXXMODULES.md [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+        // More scaffolding.
+        extern int __Cling_Autoloading_Map;
+        namespace foo{struct __attribute__((annotate("$clingAutoload$Foo.h"))) bar;}
+        struct __attribute__((annotate("$clingAutoload$Foo.h"))) S;
+        // More initialization scaffolding.
+        /*restore parsing state*/ baz1 /*store parsing state*/
+        #include <Foo.h>/*restore parsing state*/;
+```
+
+To recap, unfortunately, ROOT PCH is not extendable; ROOTMAP require a lot of
```
Comment (Axel-Naumann) [inline README/README.CXXMODULES.md]: "require*s*"

File: README/README.CXXMODULES.md [hunk 6]
```diff
@@ -0,0 +1,486 @@
...
+maintenance and goes on a very untested codepath, while RDICT has a very limited
+scope. The three features require a lot of mechanisms to work together and the
+corner cases are very many. The interaction between some of the features are
```
Comment (Axel-Naumann) [inline README/README.CXXMODULES.md]: How is the interaction not well defined? Reality contradicts, also there's no need to bash the present to show improvements.

File: README/README.CXXMODULES.md [hunk 7]
```diff
@@ -0,0 +1,486 @@
...
+As always, ROOT is (mostly) API and ABI compatible. C++ Modules-aware ROOT is no
+different. There are several differences which can be noticed:
+  * \*modulemap files in $ROOTSYS/include -- those files are used by rootcling to
+  put a set of header files in a single pcm file. For example, all related
+  headers of *libGeom* are persisted in *Geom.pcm*. There are a few notable
+  examples, which are specific to the way we build ROOT. In certain cases we
+  want some header files to be compiled within C context or with RTTI on/off.
+  That's mostly for bootstrapping ROOT (aka rootcling stage1).
+  * modulemap.overlay.yaml -- automatically created virtual filesystem overlay
+  file. This file contains introduces C++ Modules for third-party dependencies.
```
Comment (Axel-Naumann) [inline README/README.CXXMODULES.md]: "contains introduces"

Comment (Axel-Naumann) [inline README/README.CXXMODULES.md]: for *ROOT's* third-party dependencies. I first understood it's for e.g. CMSSW, where ROOT is a third-party dependency.

File: README/README.CXXMODULES.md [hunk 8]
```diff
@@ -0,0 +1,486 @@
...
+  For example, to 'modularize' glibc for ROOT we need to place a modulemap
```
Comment (Axel-Naumann) [inline README/README.CXXMODULES.md]: *would* need to place. We don't actually need to.

File: README/README.CXXMODULES.md [hunk 9]
```diff
@@ -0,0 +1,486 @@
...
+  file in (usually) `/usr/include`. This folder is not writable on many
+  platforms. The vfs file tells the compiler to pretend there is a file at a
+  specific location. This way we 'mount' `/usr/include/module.modulemap`
+  non-invasively. The reasons why we need to extend the C++ modules support
+  beyond ROOT is described bellow.
+  * rootcling creates a new binary artifact *Name.pcm* after the library name --
+  this is a temporary solution for the current technology preview. Once we
+  advance further the implementation we will only create Name.pcm without the
+  other 2 artifacts.
```
Comment (Axel-Naumann) [inline README/README.CXXMODULES.md]: Add: "At a final stage, ROOT might be able to integrate the Name.pcm with the shared library itself."

File: README/README.CXXMODULES.md [hunk 10]
```diff
@@ -0,0 +1,486 @@
...
+  * Preloads all \*pcm files at start up time -- this currently is the only
+  remaining bottleneck which introduces a relatively small performance overhead
+  described bellow.
```
Comment (Axel-Naumann) [inline README/README.CXXMODULES.md]: Clarify that this is a bottleneck for ROOT's startup time! And will be dwarfed by e.g. frameworks' startup and header parsing time.

File: README/README.CXXMODULES.md [hunk 11]
```diff
@@ -0,0 +1,486 @@
...
+  * Improved correctness in number of cases -- in a few cases ROOT is more
+  correct. In particular, when resolving global variables and function
+  declarations which are not part of the ROOT PCH.
```
Comment (Axel-Naumann) [inline README/README.CXXMODULES.md]: Even if the function decl is in the ROOT PCH we cannot load the library.

It is still incorrect to state "improved correctness"; nothing in your `gMinuit` example is broken. Instead, preloading modules extends the autoloading feature. Which is fantastic - but again you don't need to exaggerate on the past to show how nice PCMs are!

File: README/README.CXXMODULES.md [hunk 12]
```diff
@@ -0,0 +1,486 @@
...
+  * Enhanced symbol resolution mechanisms, bloom filters -- standard ROOT relies
+  on information in ROOTMAP files to react when the llvm JIT issues an
+  unresolved symbol callback. C++ Modules-aware ROOT relies on a behavior much
+  closer to the standard linker behavior. In particular, we start searching on
+  the LD_LIBRARY_PATH descending to the system libraries. The algorithm in a
+  very efficient because it uses bloom filters. This in turn allows ROOT symbol
```
Comment (Axel-Naumann) [inline README/README.CXXMODULES.md]: Wikipedia link to bloom filters?

File: README/README.CXXMODULES.md [hunk 13]
```diff
@@ -0,0 +1,486 @@
...
+  remaining bottleneck which introduces a relatively small performance overhead
+  described bellow.
+  * Improved correctness in number of cases -- in a few cases ROOT is more
+  correct. In particular, when resolving global variables and function
+  declarations which are not part of the ROOT PCH.
+  * Enhanced symbol resolution mechanisms, bloom filters -- standard ROOT relies
+  on information in ROOTMAP files to react when the llvm JIT issues an
+  unresolved symbol callback. C++ Modules-aware ROOT relies on a behavior much
+  closer to the standard linker behavior. In particular, we start searching on
+  the LD_LIBRARY_PATH descending to the system libraries. The algorithm in a
```
Comment (Axel-Naumann) [inline README/README.CXXMODULES.md]: "The algorithm in a very efficient"?

File: README/README.CXXMODULES.md (hunk 6 above)
Comment (vgvassilev) [reply README/README.CXXMODULES.md]: For example, the rootmap files require reading of forward decls at startup, then loading of the library with annotated forward decls then possibly loading definitions. This require (a) a bunch of callbacks to work together; (b) patches in the compiler introducing non-standard C++; (c) demangling symbol information to guess which is the declaration; (d) guessing the header file from the name of the identifier. That’s for rootmaps and it is pretty ugly already. Rdicts and rootmap code interact mostly in TCling::RegisterModule which is pretty hard to modify and maintain in the moment. Rdict and pch interact in rootcling, where we need to keep track of what we put in the pch and write an empty rdict.

The interacting features are breaking design and introducing layering violations.

File: README/README.CXXMODULES.md (hunk 7 above)
Comment (vgvassilev) [reply README/README.CXXMODULES.md]: Good point I will fix it.

File: README/README.CXXMODULES.md (hunk 11 above)
Comment (vgvassilev) [reply README/README.CXXMODULES.md]: It is hard to draw the line. For me `gMinuit` should behave the same as `gROOT`, `gInterpreter` and alike. Currently it does not and thus, to me, this is incorrect.

File: README/README.CXXMODULES.md (hunk 5 above)
Comment (vgvassilev) [reply README/README.CXXMODULES.md]: Done

File: README/README.CXXMODULES.md (hunk 6 above)
Comment (vgvassilev) [reply README/README.CXXMODULES.md]: Reworded to " The interaction between some of the features often
break design and introduce layering violations."

File: README/README.CXXMODULES.md (hunk 7 above)
Comment (vgvassilev) [reply README/README.CXXMODULES.md]: Fixed.

File: README/README.CXXMODULES.md (hunk 8 above)
Comment (vgvassilev) [reply README/README.CXXMODULES.md]: We do ;) but fixed.

File: README/README.CXXMODULES.md (hunk 9 above)
Comment (vgvassilev) [reply README/README.CXXMODULES.md]: Ok... done.

File: README/README.CXXMODULES.md (hunk 10 above)
Comment (vgvassilev) [reply README/README.CXXMODULES.md]: Done.

File: README/README.CXXMODULES.md (hunk 13 above)
Comment (vgvassilev) [reply README/README.CXXMODULES.md]: Fixed.

File: README/README.CXXMODULES.md (hunk 12 above)
Comment (vgvassilev) [reply README/README.CXXMODULES.md]: Done.
